cpp/src/binaryop/jit/core/parser.h

```diff
@@ -0,0 +1,507 @@
+#ifndef GDF_BINARY_OPERATION_JIT_CORE_PARSER_H
+#define GDF_BINARY_OPERATION_JIT_CORE_PARSER_H
+
+#include <cctype>
+#include <map>
+#include <string>
+#include <vector>
+
+constexpr char percent_escape[] = "_";
```

Comment:
Why is all the code in a header as inline functions? Suspect this would add to compile time when included from multiple files.  Suggest you put the implementation in a .cpp and provide a header. No need for these long functions to be inlined.

Reply:
I have fixed all CUDA related issues you mentioned.